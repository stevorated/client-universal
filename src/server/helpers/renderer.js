import React from 'react'
import { renderToString } from 'react-dom/server'
import { Helmet } from 'react-helmet'
import { StaticRouter } from 'react-router'
import { Provider } from 'react-redux'
import { ApolloProvider } from 'react-apollo'
import serialize from 'serialize-javascript'
import { ServerStyleSheet } from 'styled-components'
import Layout from '../../shared/Routes/Layout'
import { config } from '@fortawesome/fontawesome-svg-core'
config.autoAddCss = false

const { API_BASE } = process.env
const linkToFavicon = `${API_BASE}/public_images/favicon.png`
export default function renderer(req, store, client, context) {

  const App = (
    <Provider store={store}>
      <ApolloProvider client={client}>
        <StaticRouter location={req.url} context={context}>
          <Layout />
        </StaticRouter>
      </ApolloProvider>
    </Provider>
  )

  const sheet = new ServerStyleSheet()
  const content = renderToString(sheet.collectStyles(App))
  const styles = sheet.getStyleTags()
  
  const helmet = Helmet.renderStatic()

  return `
  <!DOCTYPE html>
    <html>
      <head>
      <meta charset="UTF-8">
      <meta name="viewport" content="width=device-width, initial-scale=1.0">
      <meta name="840430c5e70010722d679edee63ea147c19430e1" content="e73a6ac6dc339154d6038998ef2574d65b12bf5e" />
      <script type="text/javascript">
        WebFontConfig = {
          google: { families: [ 'Sigmar One::latin' ] }
        };
        (function() {
          var wf = document.createElement('script');
          wf.src = ('https:' == document.location.protocol ? 'https' : 'http') +
            '://ajax.googleapis.com/ajax/libs/webfont/1/webfont.js';
          wf.type = 'text/javascript';
          wf.async = 'true';
          var s = document.getElementsByTagName('script')[0];
          s.parentNode.insertBefore(wf, s);
        })(); </script>
      <link href="https://fonts.googleapis.com/css?family=Sigmar+One&display=swap" rel="stylesheet" />
      <link href="https://use.fontawesome.com/releases/v5.9.0/css/svg-with-js.css" rel="stylesheet" />
      ${helmet.title.toString()}
      ${helmet.meta.toString()}
      <link rel="icon" href="${linkToFavicon}" type="image/png"/>
      <link rel="stylesheet" href="/styles/main.css"/>
      ${styles}
      </head>
      <body>
        <div id="root">${content}</div>
        <div id="portal"></div>
        <script>
          window.__INITIAL_STATE__ = ${serialize(store.getState())}
        </script>
        <script>
          window.__APOLLO_STATE__ = ${serialize(client.extract())}
        </script>
        <script src="/bundle.js"></script>
      </body>
    </html>
  
  `
}
