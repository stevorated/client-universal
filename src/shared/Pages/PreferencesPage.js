import React, { Component } from 'react'
import { Container, Row, Col } from 'reactstrap'
import styled from 'styled-components'
import { connect } from 'react-redux'
import { HelmetComponent } from '../Components'
import { fetchFeed, clearFeed } from '../Store/actions'
import requireAuth from '../HOC/requireAuth'
import checkLoggedIn from '../HOC/checkLoggedIn'
import Menu from '../Routes/Menu'

import { mediaQs, mediaQueries } from '../Utils'

class PreferencesPage extends Component {

  constructor (props) {
    super(props)
    this.title = 'Prefs'
    this.state = {
      leaveClass: 'animated fadeOutUp'
    }
  }
  componentWillUnmount = () => {
    
  }

  render() {
    return(
      <Row className="text-center">
        <HelmetComponent pageTitle={this.title} ogTitle={this.title} />
        <FloatLeft lg="3">
          <Menu />
        </FloatLeft>
        <Col lg="6" className="offset-lg-3 order-3 order-lg-2 animated fadeIn" >
          <h1>PREFS PAGE</h1>
        </Col>
        <Col lg="3" className="order-2 order-lg-3 mt-2 animated fadeIn">
        </Col>
      </Row>
    )
  } 
}

function mapStateToProps({ users, posts, feed }) {
  return { users, posts, feed }
}

export default {
  component: connect(mapStateToProps, {fetchFeed, clearFeed})(checkLoggedIn(requireAuth(PreferencesPage))),
  loadData: ({ dispatch }) => dispatch(fetchFeed())
}
const FloatLeft = styled(Col)`
  position: static!important;
  top: 3.5rem;
  left: 0rem;
  ${mediaQueries.lg`
  position: fixed!important;
  `}
  `


