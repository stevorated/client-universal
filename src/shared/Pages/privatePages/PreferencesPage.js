import React, { Component } from "react"
import { Row, Col } from "reactstrap"
import styled from "styled-components"
import { connect } from "react-redux"
import { HelmetComponent } from "../../Components"
import requireAuth from "../../HOC/requireAuth"
import checkLoggedIn from "../../HOC/checkLoggedIn"
import Menu from "../../Routes/Menu"

import { mediaQueries } from "../../Utils"

export class PreferencesPage extends Component {
  constructor(props) {
    super(props)
    this.title = "Prefs"
  }

  render() {
    return (
      <Row data-test="main-div" className="text-center">
        <HelmetComponent
          data-test="helmet"  
          pageTitle={this.title}
          ogTitle={this.title}
        />
        <FloatLeft data-test="leftCol" lg="3">
          <Menu />
        </FloatLeft>
        <Col
          lg="6"
          data-test="mainCol"
          className="offset-lg-3 order-3 order-lg-2 animated fadeIn"
        >
          <h1>PREFS PAGE</h1>
        </Col>
        <Col
          lg="3"
          data-test="rightCol"
          className="order-2 order-lg-3 mt-2 animated fadeIn"
        />
      </Row>
    )
  }
}

function mapStateToProps({ users, posts, feed }) {
  return { users, posts, feed }
}

export default {
  component: connect(
    mapStateToProps,
    {}
  )(checkLoggedIn(requireAuth(PreferencesPage)))
}
const FloatLeft = styled(Col)`
  position: static !important;
  top: 3.5rem;
  left: 0rem;
  ${mediaQueries.lg`
  position: fixed!important;
  `}
`
