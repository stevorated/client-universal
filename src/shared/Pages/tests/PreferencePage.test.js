import React from 'react'
import { BrowserRouter } from 'react-router-dom'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import configureStore from 'redux-mock-store'

import { PreferencesPage } from '../PreferencesPage'
import { findByTestAttr } from '../../../tests/utils'
const initialState = {}
const mockStore = configureStore()
let component, store, tree;

const setupShallowRender = (props = {}) => {
  return shallow(<PreferencesPage />)
}

describe('<PreferencesPage />', () => {
  beforeEach(() => {  //creates the store with any initial state or middleware needed  
    store = mockStore(initialState)
    component = setupShallowRender()
    tree = renderer.create(
      <BrowserRouter>
        <PreferencesPage store={store}/>
      </BrowserRouter>)
   })
  it('matches Snapshot', ()=> {
    expect(tree).toMatchSnapshot()
  })
  it('Should render witout errors', ()=> {
    const mainDiv = findByTestAttr(component,'testShallowMain')
    const mainCol = findByTestAttr(component,'mainCol')
    const leftCol = findByTestAttr(component,'leftCol')
    const rightCol = findByTestAttr(component,'rightCol')
    const helmet = findByTestAttr(component,'helmet')

    expect(mainDiv.length).toBe(1)
    expect(mainCol.length).toBe(1)
    expect(leftCol.length).toBe(1)
    expect(rightCol.length).toBe(1)
    expect(helmet.length).toBe(1)
  })
  it('Should render witout errors', ()=> {
    const mainDiv = findByTestAttr(component,'testShallowMain')
    const mainCol = findByTestAttr(component,'mainCol')
    const leftCol = findByTestAttr(component,'leftCol')
    const rightCol = findByTestAttr(component,'rightCol')
    const helmet = findByTestAttr(component,'helmet')

    expect(mainDiv.length).toBe(1)
    expect(mainCol.length).toBe(1)
    expect(leftCol.length).toBe(1)
    expect(rightCol.length).toBe(1)
    expect(helmet.length).toBe(1)
  })
  it('Should render witout errors', ()=> {
    const mainDiv = findByTestAttr(component,'testShallowMain')
    const mainCol = findByTestAttr(component,'mainCol')
    const leftCol = findByTestAttr(component,'leftCol')
    const rightCol = findByTestAttr(component,'rightCol')
    const helmet = findByTestAttr(component,'helmet')

    expect(mainDiv.length).toBe(1)
    expect(mainCol.length).toBe(1)
    expect(leftCol.length).toBe(1)
    expect(rightCol.length).toBe(1)
    expect(helmet.length).toBe(1)
  })
  it('Should render Cols components witout errors', ()=> {

    const mainCol = findByTestAttr(component,'mainCol')
    const leftCol = findByTestAttr(component,'leftCol')
    const rightCol = findByTestAttr(component,'rightCol')

    expect(mainCol.length).toBe(1)
    expect(leftCol.length).toBe(1)
    expect(rightCol.length).toBe(1)

  })
  it('Should render helmet component witout errors', ()=> {

    const helmet = findByTestAttr(component,'helmet')

    expect(helmet.length).toBe(1)
  })
})