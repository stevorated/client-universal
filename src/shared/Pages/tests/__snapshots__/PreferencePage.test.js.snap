// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PreferencesPage /> matches Snapshot 1`] = `
.c1 {
  color: #c9783e;
  background: #0A090C;
  opacity: .5;
  padding: 0.5rem;
  box-shadow: 1px 1px 3px rgba(0,0,0,0.12),0 1px 2px rgba(0,0,0,0.24);
  -webkit-transition: all 3s ease;
  transition: all 3s ease;
  -webkit-transition: all .5s ease;
  transition: all .5s ease;
}

.c1:hover {
  box-shadow: 3px 3px 6px rgba(0,0,0,0.16),0 3px 6px rgba(0,0,0,0.23);
}

.c3 {
  padding: 2px 4px;
  border: none;
  background: black;
  cursor: pointer;
  list-style: none;
  display: block;
  margin: .5rem .2rem .2rem 0rem;
  -webkit-transition: all .5s ease;
  transition: all .5s ease;
}

.c3:hover {
  background: #c9783e;
  color: black;
}

.c3:active {
  background: none!important;
}

.c4 {
  color: #c9783e;
  display: block;
}

.c4:active {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c4:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #0A090C;
}

.c2 {
  display: block;
}

.c0 {
  position: static !important;
  top: 3.5rem;
  left: 0rem;
}

@media (min-width:992px) {
  .c1 {
    border-radius: .3rem;
  }
}

@media (min-width:992px) {
  .c3 {
    border: none;
    background: none;
  }
}

@media (min-width:992px) {
  .c0 {
    position: fixed!important;
  }
}

<div
  className="text-center row"
  data-test="main-div"
>
  <div />
  <div
    className="c0 col-lg-3"
    data-test="leftCol"
  >
    <div
      className="text-center mb-0 c1"
      style={
        Object {
          "borderRadius": "none",
        }
      }
    >
      <h3
        className="header-4 mt-3"
        style={
          Object {
            "color": "#c9783e",
            "fontWeight": "600",
          }
        }
      >
        Menu
      </h3>
      <hr
        className="lead mb-3"
        style={
          Object {
            "borderColor": "#c9783e",
            "borderWidth": "2px",
            "color": "#c9783e",
          }
        }
      />
      <div
        className="container"
      >
        <div
          className="row"
        >
          <div
            className="p-0 col-6 col-lg-12"
          >
            <ul
              className="p-0 m-0 pb-2 c2"
            >
              <li
                className="text-center pl-3 m-2 c3"
              >
                <a
                  className="text-left py-1 c4"
                  href="/event-board"
                  onClick={[Function]}
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-anchor fa-w-18 fa-lg mx-3"
                    data-icon="anchor"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    style={Object {}}
                    viewBox="0 0 576 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12.971 352h32.394C67.172 454.735 181.944 512 288 512c106.229 0 220.853-57.38 242.635-160h32.394c10.691 0 16.045-12.926 8.485-20.485l-67.029-67.029c-4.686-4.686-12.284-4.686-16.971 0l-67.029 67.029c-7.56 7.56-2.206 20.485 8.485 20.485h35.146c-20.29 54.317-84.963 86.588-144.117 94.015V256h52c6.627 0 12-5.373 12-12v-40c0-6.627-5.373-12-12-12h-52v-5.47c37.281-13.178 63.995-48.725 64-90.518C384.005 43.772 341.605.738 289.37.01 235.723-.739 192 42.525 192 96c0 41.798 26.716 77.35 64 90.53V192h-52c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h52v190.015c-58.936-7.399-123.82-39.679-144.117-94.015h35.146c10.691 0 16.045-12.926 8.485-20.485l-67.029-67.029c-4.686-4.686-12.284-4.686-16.971 0L4.485 331.515C-3.074 339.074 2.28 352 12.971 352zM288 64c17.645 0 32 14.355 32 32s-14.355 32-32 32-32-14.355-32-32 14.355-32 32-32z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                  <span>
                    Event Board
                  </span>
                </a>
              </li>
              <li
                className="text-center pl-3 m-2 c3"
              >
                <a
                  className="text-left py-1 c4"
                  href="/event-feed"
                  onClick={[Function]}
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-utensils fa-w-13 fa-lg mx-3"
                    data-icon="utensils"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    style={Object {}}
                    viewBox="0 0 416 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M207.9 15.2c.8 4.7 16.1 94.5 16.1 128.8 0 52.3-27.8 89.6-68.9 104.6L168 486.7c.7 13.7-10.2 25.3-24 25.3H80c-13.7 0-24.7-11.5-24-25.3l12.9-238.1C27.7 233.6 0 196.2 0 144 0 109.6 15.3 19.9 16.1 15.2 19.3-5.1 61.4-5.4 64 16.3v141.2c1.3 3.4 15.1 3.2 16 0 1.4-25.3 7.9-139.2 8-141.8 3.3-20.8 44.7-20.8 47.9 0 .2 2.7 6.6 116.5 8 141.8.9 3.2 14.8 3.4 16 0V16.3c2.6-21.6 44.8-21.4 48-1.1zm119.2 285.7l-15 185.1c-1.2 14 9.9 26 23.9 26h56c13.3 0 24-10.7 24-24V24c0-13.2-10.7-24-24-24-82.5 0-221.4 178.5-64.9 300.9z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                  <span>
                    Event Feed
                  </span>
                </a>
              </li>
              <li
                className="text-center pl-3 m-2 c3"
              >
                <a
                  className="text-left py-1 c4"
                  href="/feed"
                  onClick={[Function]}
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-cookie-bite fa-w-16 fa-lg mx-3"
                    data-icon="cookie-bite"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    style={Object {}}
                    viewBox="0 0 512 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M510.52 255.82c-69.97-.85-126.47-57.69-126.47-127.86-70.17 0-127-56.49-127.86-126.45-27.26-4.14-55.13.3-79.72 12.82l-69.13 35.22a132.221 132.221 0 0 0-57.79 57.81l-35.1 68.88a132.645 132.645 0 0 0-12.82 80.95l12.08 76.27a132.521 132.521 0 0 0 37.16 72.96l54.77 54.76a132.036 132.036 0 0 0 72.71 37.06l76.71 12.15c27.51 4.36 55.7-.11 80.53-12.76l69.13-35.21a132.273 132.273 0 0 0 57.79-57.81l35.1-68.88c12.56-24.64 17.01-52.58 12.91-79.91zM176 368c-17.67 0-32-14.33-32-32s14.33-32 32-32 32 14.33 32 32-14.33 32-32 32zm32-160c-17.67 0-32-14.33-32-32s14.33-32 32-32 32 14.33 32 32-14.33 32-32 32zm160 128c-17.67 0-32-14.33-32-32s14.33-32 32-32 32 14.33 32 32-14.33 32-32 32z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                  <span>
                    Post Feed
                  </span>
                </a>
              </li>
              <li
                className="text-center pl-3 m-2 c3"
              >
                <a
                  className="text-left py-1 c4"
                  href="/calander"
                  onClick={[Function]}
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-calendar-alt fa-w-14 fa-lg mx-3"
                    data-icon="calendar-alt"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    style={Object {}}
                    viewBox="0 0 448 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V192H0v272zm320-196c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zm0 128c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zM192 268c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zm0 128c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zM64 268c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12H76c-6.6 0-12-5.4-12-12v-40zm0 128c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12H76c-6.6 0-12-5.4-12-12v-40zM400 64h-48V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H160V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H48C21.5 64 0 85.5 0 112v48h448v-48c0-26.5-21.5-48-48-48z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                  <span>
                    Calander
                  </span>
                </a>
              </li>
            </ul>
          </div>
          <div
            className="p-0 pb-2 col-6 col-lg-12"
          >
            <ul
              className="p-0 m-0 c2"
            >
              <li
                className="text-center pl-3 m-2 c3"
              >
                <a
                  className="text-left py-1 c4"
                  href="/my-profile"
                  onClick={[Function]}
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-user-alt fa-w-16 fa-lg mx-3"
                    data-icon="user-alt"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    style={Object {}}
                    viewBox="0 0 512 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M256 288c79.5 0 144-64.5 144-144S335.5 0 256 0 112 64.5 112 144s64.5 144 144 144zm128 32h-55.1c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16H128C57.3 320 0 377.3 0 448v16c0 26.5 21.5 48 48 48h416c26.5 0 48-21.5 48-48v-16c0-70.7-57.3-128-128-128z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                  <span>
                    Profile
                  </span>
                </a>
              </li>
              <li
                className="text-center pl-3 m-2 c3"
              >
                <a
                  className="text-left py-1 c4"
                  href="/engage-gauge "
                  onClick={[Function]}
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-brain fa-w-18 fa-lg mx-3"
                    data-icon="brain"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    style={Object {}}
                    viewBox="0 0 576 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M208 0c-29.9 0-54.7 20.5-61.8 48.2-.8 0-1.4-.2-2.2-.2-35.3 0-64 28.7-64 64 0 4.8.6 9.5 1.7 14C52.5 138 32 166.6 32 200c0 12.6 3.2 24.3 8.3 34.9C16.3 248.7 0 274.3 0 304c0 33.3 20.4 61.9 49.4 73.9-.9 4.6-1.4 9.3-1.4 14.1 0 39.8 32.2 72 72 72 4.1 0 8.1-.5 12-1.2 9.6 28.5 36.2 49.2 68 49.2 39.8 0 72-32.2 72-72V64c0-35.3-28.7-64-64-64zm368 304c0-29.7-16.3-55.3-40.3-69.1 5.2-10.6 8.3-22.3 8.3-34.9 0-33.4-20.5-62-49.7-74 1-4.5 1.7-9.2 1.7-14 0-35.3-28.7-64-64-64-.8 0-1.5.2-2.2.2C422.7 20.5 397.9 0 368 0c-35.3 0-64 28.6-64 64v376c0 39.8 32.2 72 72 72 31.8 0 58.4-20.7 68-49.2 3.9.7 7.9 1.2 12 1.2 39.8 0 72-32.2 72-72 0-4.8-.5-9.5-1.4-14.1 29-12 49.4-40.6 49.4-73.9z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                  <span>
                    Engage Page
                  </span>
                </a>
              </li>
              <li
                className="text-center pl-3 m-2 c3"
              >
                <a
                  className="text-left py-1 c4"
                  href="/settings "
                  onClick={[Function]}
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-wrench fa-w-16 fa-lg mx-3"
                    data-icon="wrench"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    style={Object {}}
                    viewBox="0 0 512 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M507.73 109.1c-2.24-9.03-13.54-12.09-20.12-5.51l-74.36 74.36-67.88-11.31-11.31-67.88 74.36-74.36c6.62-6.62 3.43-17.9-5.66-20.16-47.38-11.74-99.55.91-136.58 37.93-39.64 39.64-50.55 97.1-34.05 147.2L18.74 402.76c-24.99 24.99-24.99 65.51 0 90.5 24.99 24.99 65.51 24.99 90.5 0l213.21-213.21c50.12 16.71 107.47 5.68 147.37-34.22 37.07-37.07 49.7-89.32 37.91-136.73zM64 472c-13.25 0-24-10.75-24-24 0-13.26 10.75-24 24-24s24 10.74 24 24c0 13.25-10.75 24-24 24z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                  <span>
                    Settings
                  </span>
                </a>
              </li>
              <li
                className="text-center pl-3 m-2 c3"
              >
                <a
                  className="text-left py-1 c4"
                  href="/privacy"
                  onClick={[Function]}
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-air-freshener fa-w-12 fa-lg mx-3"
                    data-icon="air-freshener"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    style={Object {}}
                    viewBox="0 0 384 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M378.94 321.41L284.7 224h49.22c15.3 0 23.66-16.6 13.86-27.53L234.45 69.96c3.43-6.61 5.55-14 5.55-21.96 0-26.51-21.49-48-48-48s-48 21.49-48 48c0 7.96 2.12 15.35 5.55 21.96L36.22 196.47C26.42 207.4 34.78 224 50.08 224H99.3L5.06 321.41C-6.69 333.56 3.34 352 21.7 352H160v32H48c-8.84 0-16 7.16-16 16v96c0 8.84 7.16 16 16 16h288c8.84 0 16-7.16 16-16v-96c0-8.84-7.16-16-16-16H224v-32h138.3c18.36 0 28.39-18.44 16.64-30.59zM192 31.98c8.85 0 16.02 7.17 16.02 16.02 0 8.84-7.17 16.02-16.02 16.02S175.98 56.84 175.98 48c0-8.85 7.17-16.02 16.02-16.02zM304 432v32H80v-32h224z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                  <span>
                    Privacy
                  </span>
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="offset-lg-3 order-3 order-lg-2 animated fadeIn col-lg-6"
    data-test="mainCol"
  >
    <h1>
      PREFS PAGE
    </h1>
  </div>
  <div
    className="order-2 order-lg-3 mt-2 animated fadeIn col-lg-3"
    data-test="rightCol"
  />
</div>
`;
